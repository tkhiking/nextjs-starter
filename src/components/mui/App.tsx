// https://github.com/mui-org/material-ui/tree/master/examples/nextjs

import App from 'next/app';
import React from 'react';

import CommonHead from './CommonHead';
import CommonProviders from './CommonProviders';

/* eslint-disable react/jsx-props-no-spreading */
class MuiApp extends App {
  componentDidMount(): void {
    // Remove the server-side injected CSS.
    const jssStyles = document.querySelector('#jss-server-side');
    if (jssStyles?.parentElement) {
      // eslint-disable-next-line unicorn/prefer-node-remove
      jssStyles.parentElement.removeChild(jssStyles);
    }
  }

  render(): React.ReactElement {
    const { Component, pageProps } = this.props;

    return (
      <CommonHead>
        <CommonProviders>
          <Component {...pageProps} />
        </CommonProviders>
      </CommonHead>
    );
  }
}
/* eslint-enable react/jsx-props-no-spreading */

export default MuiApp;

// const didMountRef = React.useRef(false);
// React.useLayoutEffect(() => {
//   if (!didMountRef.current) {
//     // Remove the server-side injected CSS.
//     const jssStyles = document.querySelector('#jss-server-side');
//     if (jssStyles?.parentElement) {
//       jssStyles.parentElement.removeChild(jssStyles);
//     }
//     didMountRef.current = true;
//   }
// });
