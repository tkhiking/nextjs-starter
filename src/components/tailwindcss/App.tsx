/* eslint-disable react/jsx-props-no-spreading */

import { AppProps } from 'next/app';
import React from 'react';

import CommonProviders from './CommonProviders';

function TailwindApp({ Component, pageProps }: AppProps): React.ReactElement {
  return (
    <CommonProviders>
      <Component {...pageProps} />
    </CommonProviders>
  );
}

export default TailwindApp;
