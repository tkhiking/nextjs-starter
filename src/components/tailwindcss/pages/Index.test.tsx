/* eslint-env jest */
/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';

import App from './Index';

describe('With React Testing Library', () => {
  it('Shows "Welcome to Next.js starter."', () => {
    const { getByText } = render(<App />);

    expect(getByText('Welcome to Next.js starter.')).not.toBeNull();
  });
});

describe('With React Testing Library Snapshot', () => {
  it('Should match Snapshot', () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });
});

/* eslint-enable import/no-extraneous-dependencies */
