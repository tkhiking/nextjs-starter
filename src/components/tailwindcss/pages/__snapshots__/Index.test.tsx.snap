// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With React Testing Library Snapshot Should match Snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-background"
  >
    <div
      class="flex flex-col  container mx-auto min-h-screen  bg-surface shadow-xl"
    >
      <header
        class="flex flex-col  fixed container h-16 z-10 p-2  bg-surface"
      >
        <div
          class="flex  flex-1 relative"
        >
          <h1
            class="flex-1 my-auto px-2  text-lg text-on-surface font-medium truncate"
          >
            <a
              href="/"
            >
              Next.js starter
            </a>
          </h1>
          <div
            class="my-auto"
            id="account-menu"
          >
            <label
              class="tk-button-circle px-2"
              data-test-id="account-icon"
              for="tk-account-menu-checkbox"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
            </label>
            <input
              class="tk-account-menu-checkbox"
              id="tk-account-menu-checkbox"
              type="checkbox"
            />
            <label
              class="tk-account-menu-overlay"
              for="tk-account-menu-checkbox"
            />
            <nav
              class="tk-account-menu-nav"
            >
              <a
                class="tk-menu-nav-item tk-button-ripple"
                href="/login"
              >
                Log in
              </a>
              <a
                class="tk-menu-nav-item tk-button-ripple"
                href="/signup"
              >
                Sign up
              </a>
            </nav>
          </div>
          <div
            class="my-auto"
            id="nav-menu"
          >
            <label
              class="tk-menu-button tk-button-circle"
              data-test-id="menu-icon"
              for="tk-menu-checkbox"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </label>
            <input
              class="tk-menu-checkbox"
              id="tk-menu-checkbox"
              type="checkbox"
            />
            <label
              class="tk-menu-overlay"
              for="tk-menu-checkbox"
            />
            <nav
              class="tk-menu-nav"
            >
              <a
                class="tk-menu-nav-item tk-button-ripple"
                href="/"
              >
                Home
              </a>
              <a
                class="tk-menu-nav-item tk-button-ripple"
                href="/users"
              >
                Users
              </a>
            </nav>
          </div>
        </div>
        <hr />
      </header>
      <div
        class="mt-16"
      />
      <div
        class="flex  flex-1"
      >
        <nav
          class="hidden md:block flex-shrink-0 w-64  border-r border-on-surface-border"
        >
          <a
            class="tk-menu-nav-item tk-button-ripple"
            href="/"
          >
            Home
          </a>
          <a
            class="tk-menu-nav-item tk-button-ripple"
            href="/users"
          >
            Users
          </a>
        </nav>
        <main
          class="flex  flex-1"
        >
          <div
            class="p-2"
          >
            <p
              class="text-4xl text-on-surface"
            >
              Welcome to Next.js starter.
            </p>
            <button
              class="tk-button tk-button-contained tk-button-ripple"
              type="button"
            >
              Contained button
            </button>
            <div
              class="bg-surface"
            >
              <p
                class="text-on-surface"
              >
                <span>
                  text-on-surface
                </span>
              </p>
              <p
                class="text-on-surface-light"
              >
                <span>
                  text-on-surface-light
                </span>
              </p>
              <p
                class="text-on-surface-dark"
              >
                <span>
                  text-on-surface-dark
                </span>
              </p>
              <p
                class="text-on-surface-text-high-emphasis"
              >
                <span>
                  text-on-surface-text-high-emphasis
                </span>
              </p>
              <p
                class="text-on-surface-text-medium-emphasis"
              >
                <span>
                  text-on-surface-text-medium-emphasis
                </span>
              </p>
              <p
                class="text-on-surface-text-disabled"
              >
                <span>
                  text-on-surface-text-disabled
                </span>
              </p>
            </div>
            <div
              class="bg-background"
            >
              <p
                class="text-on-background"
              >
                <span>
                  text-on-background
                </span>
              </p>
              <p
                class="text-on-background-text-high-emphasis"
              >
                <span>
                  text-on-background-text-high-emphasis
                </span>
              </p>
              <p
                class="text-on-background-text-medium-emphasis"
              >
                <span>
                  text-on-background-text-medium-emphasis
                </span>
              </p>
              <p
                class="text-on-background-text-disabled"
              >
                <span>
                  text-on-background-text-disabled
                </span>
              </p>
            </div>
          </div>
        </main>
      </div>
      <footer
        class="flex flex-col  h-16 p-2"
      >
        <hr />
        <p
          class="m-auto  text-on-surface"
        >
          Powered byÂ 
          <a
            class="text-primary"
            href="https://github.com/twihike"
          >
            Next.js starter
          </a>
        </p>
      </footer>
    </div>
  </div>
</DocumentFragment>
`;
